src/ui/App.tsx

```diff
@@ -0,0 +1,380 @@
+          ...prev,
+          [packageName]: 'Version is required',
+        }));
+        return;
+      }
+
+      const newVersion = new SemVer(version);
+      const currentVersion = new SemVer(
+        packages.find((p) => p.name === packageName)?.version || '0.0.0',
+      );
+
+      if (newVersion.compare(currentVersion) <= 0) {
+        setVersionErrors((prev) => ({
+          ...prev,
+          [packageName]: 'New version must be higher than current version',
+        }));
+        return;
+      }
+
+      setVersionErrors((prev) => {
+        const { [packageName]: _, ...rest } = prev;
+        return rest;
+      });
+
+      setSelections((prev) => ({
+        ...prev,
+        [packageName]: version,
+      }));
+    } catch (err) {
+      setVersionErrors((prev) => ({
+        ...prev,
+        [packageName]: 'Invalid semver version',
+      }));
+    }
+  };
+
+  const handleSelectionChange = (
+    packageName: string,
+    value: ReleaseType | '',
+  ): void => {
+    setSelections((prev) => {
+      if (value === '') {
+        const { [packageName]: _, ...rest } = prev;
+        const { [packageName]: __, ...remainingErrors } =
+          packageDependencyErrors;
+        setPackageDependencyErrors(remainingErrors);
+        return rest;
+      }
+      return {
+        ...prev,
+        [packageName]: value,
+      };
+    });
+  };
+
+  const handleSubmit = async (): Promise<void> => {
+    setIsSubmitting(true);
+    try {
+      const response = await fetch('/api/release', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(selections),
+      });
+
+      const data: {
+        status: 'success' | 'error';
+        packagesErrors?: Record<
+          string,
+          {
+            missingDependentNames: string[];
+            missingDependencies: string[];
+          }
+        >;
+        errors?: {
+          message: string | string[];
+          lineNumber?: number;
+        }[];
+      } = await response.json();
+
+      if (data.status === 'error' && data.errors) {
+        setSubmitErrors(
+          data.errors.flatMap((error) => {
+            if (Array.isArray(error.message)) {
+              return error.message;
+            }
+            return error.message;
+          }),
+        );
+      }
+
+      if (data.status === 'success') {
+        setIsSuccess(true);
+      }
+    } catch (err) {
+      const errorMessage =
+        err instanceof Error ? err.message : 'An error occurred';
+      setError(errorMessage);
+      console.error('Error submitting selections:', err);
+      alert('Failed to submit selections. Please try again.');
+    } finally {
+      setIsSubmitting(false);
+    }
+  };
+
+  const fetchChangelog = async (packageName: string): Promise<void> => {
+    setLoadingChangelogs((prev) => ({ ...prev, [packageName]: true }));
+    try {
+      const response = await fetch(`/api/changelog?package=${packageName}`);
```

Comment:
I notice that this returns the entire changelog for the package. It'd be nice if in a future we could capture and return the unreleased changes instead. I feel like those might be more useful than providing the entire changelog, because it's the unreleased changes that dictate how you want to bump a package, or whether you need to include it in the release.

Reply:
Yes I agree with this